hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/ReadableAccountStoreImpl.java

```diff
@@ -132,6 +135,33 @@ protected Account getAccountLeaf(@NonNull final AccountID id) {
                         .build());
     }
 
+    /**
+     * Returns the {@link AccountID} referenced by the given alias, whether it is a direct reference or
+     * an "indirect" reference by a proto ECDSA key whose implied EVM address is the alias.
+     *
+     * @param alias the alias to look up
+     * @return the account id referenced by the alias, or null if not found
+     */
+    private @Nullable AccountID unaliasWithEvmAddressConversionIfNeeded(@NonNull final Bytes alias) {
+        final var maybeDirectReference = aliases.get(new ProtoBytes(alias));
+        if (maybeDirectReference != null) {
+            return maybeDirectReference;
+        } else {
+            try {
+                final var protoKey = Key.PROTOBUF.parseStrict(BufferedData.wrap(alias.toByteArray()));
+                if (protoKey.hasEcdsaSecp256k1()) {
+                    final var evmAddress = recoverAddressFromPubKey(
+                            protoKey.ecdsaSecp256k1OrThrow().toByteArray());
+                    return evmAddress.length > 0 ? aliases.get(new ProtoBytes(Bytes.wrap(evmAddress))) : null;
```

Comment:
We could add a new method for recoverAddressFromPubKey that takes Bytes and returns Bytes, so you don't have to do the wrapping and unwrapping here. That would make the code a little more obvious.

Reply:
Yes, i like this, added.